content/300-guides/300-upgrade-guides/200-upgrading-versions/600-upgrading-to-prisma-5.mdx

```diff
@@ -0,0 +1,677 @@
+
+- `--preview-feature` used in `db execute`, `db seed`, and `db diff`
+- `--experimental` and `--early-access-feature` used in `migrate`
+- `--force`/`-f` used in `db push`
+- `--experimental-reintrospection` and `--clean` used in `db pull`
+
+The outdated use of `db push --force` can be replaced with the newer implementation `db push --accept-data-loss`.
+
+All other flags are from previous APIs and are no longer needed.
+
+### Removal of the <inlinecode>beforeExit</inlinecode> hook from the library engine
+
+The `beforeExit` hook has been removed from the Prisma library engine. Instead of this hook we recommend using built-in Node.js exit events.
+
+The following code with Prisma 4:
+
+```js
+const exitHandler = () => {
+  // your exit handler code
+}
+
+prisma.$on('beforeExit', exitHandler)
+```
+
+Could become:
+
+```js
+const exitHandler = () => {
+  // your exit handler code
+}
+
+process.on('exit', exitHandler)
+process.on('beforeExit', exitHandler)
+process.on('SIGINT', exitHandler)
+process.on('SIGTERM', exitHandler)
+process.on('SIGUSR2', exitHandler)
+```
+
+### Removal of deprecated <inlinecode>prisma2</inlinecode> executable
+
+When we released Prisma 2, the `prisma2` executable was used in order to differentiate from Prisma 1. In a later release, the `prisma2` cli took over the `prisma` executable name.
+
+Needless to say, the `prisma2` executable has been deprecated for some time and is now removed. If your scripts use Prisma CLI as `prisma2`, please replace it with simply `prisma`.
+
+### Removal of deprecated <inlinecode>experimentalFeatures</inlinecode> property
+
+The `previewFeatures` field of the [generator block](https://www.prisma.io/docs/reference/api-reference/prisma-schema-reference#fields-1) used to be called `experimentalFeatures`. We are finally removing that deprecated property.
+
+In Prisma 5, you will need to update references of `experimentalFeatures` to `previewFeatures` manually or use the new code action in the Prisma VSCode extension.
+
+### <inlinecode>migration-engine</inlinecode> renamed to <inlinecode>schema-engine</inlinecode>
+
+The engine responsible for commands like `prisma migrate` and `prisma db` has been renamed from `migration-engine` to `schema-engine` to better describe its use. For many users, no changes will be required. However, if you need to explicitly include or exclude this engine file for any reason, you will need to update your code references.
+
+#### Example with the Serverless Framework
+
+One example we have seen is projects using the Serverless Framework. In these instances, you will need to update any patterns that reference `migration-engine` to instead reference `schema-engine`.
+
+```yaml highlight=6;delete|7;add
+package:
+  patterns:
+    - '!node_modules/.prisma/client/libquery_engine-*'
+    - 'node_modules/.prisma/client/libquery_engine-rhel-*'
+    - '!node_modules/prisma/libquery_engine-*'
+    - '!node_modules/prisma/migration-engine-*'
```

Comment:
This line can be removed now
```suggestion
```

Reply:
no, the idea is this line is highlighted as removed and the next line is highlighted as added (see the code block header and how it's rendered)

Reply:
I'm very good at not seeing these 🙈

Reply:
It's not great for accessibility though, since nothing but color is used to encode this information.